peps/pep-0747.rst

```diff
@@ -569,19 +535,38 @@ Subtyping
 Whether a ``TypeExpr`` value can be assigned from one variable to another is
 determined by the following rules:
 
+Relationship with type
+''''''''''''''''''''''
+
 ``TypeExpr[]`` is covariant in its argument type, just like ``type[]``:
 
 -  ``TypeExpr[T1]`` is a subtype of ``TypeExpr[T2]`` iff ``T1`` is a
    subtype of ``T2``.
 -  ``type[C1]`` is a subtype of ``TypeExpr[C2]`` iff ``C1`` is a subtype
    of ``C2``.
 
-A plain ``type`` can be assigned to a plain ``TypeExpr`` but not the
-other way around:
+An unparameterized ``type`` can be assigned to an unparameterized ``TypeExpr``
+but not the other way around:
 
 -  ``type[Any]`` is assignable to ``TypeExpr[Any]``. (But not the
    other way around.)
 
+Relationship with UnionType
+'''''''''''''''''''''''''''
+
+``TypeExpr[U]`` is a subtype of ``UnionType`` iff ``U`` is a non-empty union type:
```

Comment:
Why does this need to be specified? It feels like asking type checkers to understand details of runtime implementation.

Reply:
From §"Backward Compatibility":

> As a value expression, ``X | Y`` previously had type ``UnionType`` (via :pep:`604`)
but this PEP gives it the more-precise static type ``TypeExpr[X | Y]`` 
(a subtype of ``UnionType``) while continuing to return a ``UnionType`` instance at runtime.
Preserving compability with ``UnionType`` is important because ``UnionType``
supports ``isinstance`` checks, unlike ``TypeExpr``, and existing code relies
on being able to perform those checks.

Rephrasing:

* `type.__or__` (and other methods) now have return type `TypeExpr[X | Y]` rather than `UnionType`
* Static type checkers need to treat `TypeExpr[X | Y]` as assignable to `UnionType` so that existing methods like `isinstance` which expect a `UnionType` continue to pass type checking when given a `X | Y` expression.
    * For example `isinstance('words', int | str)` needs to still pass type checking even though `int | str` is now a `TypeExpr[int | str]` and `isinstance` expects a `UnionType` as its second argument.